tests/providers/google/facebook_ads_to_gcs/operators/test_facebook_ads_to_gcs_system.py

```diff
@@ -0,0 +1,58 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+import os
+
+import mock
+import pytest
+
+from airflow.models import Connection
+from tests.providers.google.cloud.utils.gcp_authenticator import GCP_GCS_KEY
+from tests.test_utils.gcp_system_helpers import (
+    FACEBOOK_ADS_TO_GCS_DAG_FOLDER, GoogleSystemTest, provide_gcp_context,
+)
+
+CREDENTIALS_DIR = os.environ.get('CREDENTIALS_DIR', '/files/airflow-breeze-config/keys')
+FACEBOOK_KEY = 'facebook.json'
+FACEBOOK_CREDENTIALS_PATH = os.path.join(CREDENTIALS_DIR, FACEBOOK_KEY)
+CONNECTION_TYPE = os.environ.get('CONNECTION_TYPE', 'facebook_social')
+FACEBOOK_CONNECTION_ID = os.environ.get('FACEBOOK_CONNECTION_ID', 'facebook_default')
+
+
+@pytest.mark.credential_file(FACEBOOK_KEY)
+@pytest.mark.credential_file(GCP_GCS_KEY)
+@pytest.mark.system("google.facebook_ads_to_gcs")
+@pytest.mark.system("google.cloud")
+class FacebookAdsToGcsExampleDagsSystemTest(GoogleSystemTest):
+    def setUp(self):
+        super().setUp()
+        with open(FACEBOOK_CREDENTIALS_PATH) as f:
+            credentials = json.load(f)
+        conn = Connection(
+            conn_id=FACEBOOK_CONNECTION_ID,
+            conn_type=CONNECTION_TYPE,
+            extra=json.dumps(credentials)
+        )
+        self.mock_connection = {f"AIRFLOW_CONN_{conn.conn_id.upper()}": conn.get_uri()}
+
+    @provide_gcp_context(GCP_GCS_KEY)
+    def test_dag_example(self):
+        k = mock.patch.dict('os.environ', self.mock_connection)
+        k.start()
```

Comment:
Can you use decorator for mock? It prevents side effects when an exception occurs.